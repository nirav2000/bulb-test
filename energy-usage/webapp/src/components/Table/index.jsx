import React from 'react';
import PropTypes from 'prop-types';
import { formatDate } from '../../utils/DateUtil';

const Table = ({ meterReadings }) => (
  <div>
    <h2>Meter Readings</h2>
    <table>
      <tbody>
        <tr>
          <th>Date</th>
          <th>Reading</th>
          <th>Unit</th>
        </tr>
        {meterReadings.map(reading => (
          <tr key={reading.readingDate}>
            <td>{formatDate(reading.readingDate)}</td>
            <td>{reading.cumulative}</td>
            <td>{reading.unit}</td>
          </tr>
        ))}
      </tbody>
    </table>
  </div>
);

Table.propTypes = {
  meterReadings: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Table;
