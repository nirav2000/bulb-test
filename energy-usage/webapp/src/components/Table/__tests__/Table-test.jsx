import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import data from '../../data/../../data/meterReadingsSample.json';
import Table from '../index';

describe('<Table /> test', () => {
  it('Test snapshots of content', () => {
    const wrapper = shallow(<Table meterReadings={data.electricity} />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
