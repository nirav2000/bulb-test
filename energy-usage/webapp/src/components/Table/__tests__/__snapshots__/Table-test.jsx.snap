// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> test Test snapshots of content 1`] = `
<div>
  <h2>
    Meter Readings
  </h2>
  <table>
    <tbody>
      <tr>
        <th>
          Date
        </th>
        <th>
          Reading
        </th>
        <th>
          Unit
        </th>
      </tr>
      <tr
        key="2017-03-31T00:00:00.000Z"
      >
        <td>
          2017-03-31
        </td>
        <td>
          17600
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-04-30T00:00:00.000Z"
      >
        <td>
          2017-04-30
        </td>
        <td>
          17859
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-05-31T00:00:00.000Z"
      >
        <td>
          2017-05-31
        </td>
        <td>
          18102
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-06-30T00:00:00.000Z"
      >
        <td>
          2017-06-30
        </td>
        <td>
          18290
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-07-31T00:00:00.000Z"
      >
        <td>
          2017-07-31
        </td>
        <td>
          18453
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-08-31T00:00:00.000Z"
      >
        <td>
          2017-08-31
        </td>
        <td>
          18620
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-09-30T00:00:00.000Z"
      >
        <td>
          2017-09-30
        </td>
        <td>
          18782
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-10-31T00:00:00.000Z"
      >
        <td>
          2017-10-31
        </td>
        <td>
          18965
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-11-30T00:00:00.000Z"
      >
        <td>
          2017-11-30
        </td>
        <td>
          19230
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2017-12-31T00:00:00.000Z"
      >
        <td>
          2017-12-31
        </td>
        <td>
          19517
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2018-01-31T00:00:00.000Z"
      >
        <td>
          2018-01-31
        </td>
        <td>
          19827
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2018-02-28T00:00:00.000Z"
      >
        <td>
          2018-02-28
        </td>
        <td>
          20113
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2018-03-31T00:00:00.000Z"
      >
        <td>
          2018-03-31
        </td>
        <td>
          20376
        </td>
        <td>
          kWh
        </td>
      </tr>
      <tr
        key="2018-04-30T00:00:00.000Z"
      >
        <td>
          2018-04-30
        </td>
        <td>
          20610
        </td>
        <td>
          kWh
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
